app/controllers/admin/quizzs_controller.rb

```diff
@@ -0,0 +1,116 @@
+class Admin::QuizzsController < ApplicationController
+
+  before_action :authenticate_user!
+  before_action :admin_action?
+
+  before_action :set_quizz, only: [:show, :edit, :update, :destroy, :add_question, :save_question, :complete]
+  
+  before_action :set_quizzes, only: [:index, :update, :save_question]
+
+  respond_to :js, only: [:new, :edit, :create, :update, :add_question]
+
+  def index
+  end
+
+  def show
+    respond_with(@quizz)
+  end
+
+  def new
+    @quizz = Quizz.new
+  end
+
+  def edit
+    flash.now[:notice] = I18n.t('views.quizzes.warning_box.mark_answers') unless @quizz.all_marked?
+  end
+
+  def create
+    @quizz = Quizz.new(quizz_params)
+    @quizz.options['category_level'] = category_params[:options] if category_params
+    flash.now[:notice] = I18n.t('controllers.quizzes.created', :quizz => @quizz.id) if @valid = @quizz.save
+  end
+
+  def update
+    @valid = @quizz.update(quizz_params)
+
+    if question_params[:questions]
+      question_params[:questions].each do |question_id, data|
+        q =  @quizz.questions.find_or_create_by(:id => question_id)
+        q.title = data[:title]
+        update_answers(data, q)
+        @valid = q.save
```

Comment:
what if you will have 5 items in array, first 4 are invalid and then last is valid